packages/nx/src/adapter/ngcli-adapter.ts

```diff
@@ -51,10 +53,73 @@ import {
   CustomHasher,
   Executor,
   ExecutorConfig,
+  ExecutorContext,
   ExecutorsJson,
   TaskGraphExecutor,
 } from '../config/misc-interfaces';
 import { readPluginPackageJson } from '../utils/nx-plugin';
+import { WorkspaceNodeModulesArchitectHost as AngularWorkspaceNodeModulesArchitectHost } from '@angular-devkit/architect/node';
```

Comment:
This top-level import can reintroduce the issues addressed in https://github.com/nrwl/nx/pull/16595. You can wrap the `WrappedWorkspaceNodeModulesArchitectHost` class declaration in a function, so you make this import inside that function and have the consumers invoke it.